numba/core/new_boxing.py

```diff
@@ -20,18 +20,17 @@
 
 def box_np_scalars(typ, val, c, py_boxing_func, py_type):
     py_scalar = py_boxing_func(py_type, val, c)
-
-    numpy_name = c.context.insert_const_string(c.builder.module, 'numpy')
-    numpy_module = c.pyapi.import_module_noblock(numpy_name)
-    type_str = typ.name.split('np_')[1]
-
-    np_scalar_constructor = c.pyapi.object_getattr_string(numpy_module, type_str)
-    np_scalar = c.pyapi.call_function_objargs(np_scalar_constructor, (py_scalar,))
-
-    c.pyapi.decref(numpy_module)
-    c.pyapi.decref(np_scalar_constructor)
-    c.pyapi.decref(py_scalar)
-
+    with cgutils.if_likely(c.builder, cgutils.is_not_null(c.builder, py_scalar)):
+        numpy_name = c.context.insert_const_string(c.builder.module, 'numpy')
+        numpy_module = c.pyapi.import_module_noblock(numpy_name)
+        with cgutils.if_likely(c.builder, cgutils.is_not_null(c.builder, numpy_module)):
+            type_str = typ.name.split('np_')[1]
+            np_scalar_constructor = c.pyapi.object_getattr_string(numpy_module, type_str)
+            c.pyapi.decref(numpy_module)
+            with cgutils.if_likely(c.builder, cgutils.is_not_null(c.builder, np_scalar_constructor)):
+                np_scalar = c.pyapi.call_function_objargs(np_scalar_constructor, (py_scalar,))
+                c.pyapi.decref(np_scalar_constructor)
+        c.pyapi.decref(py_scalar)
```

Comment:
I'm facing an error from LLVM that I'm unable to resolve over here:

```
E               RuntimeError: Instruction does not dominate all uses!
E                 %.71 = call i8* (i8*, ...) @PyObject_CallFunctionObjArgs(i8* %.67, i8* %.61, i8* null)
E                 ret i8* %.71

/home/kc611/miniconda3/envs/type_system/lib/python3.11/site-packages/llvmlite/binding/module.py:115: RuntimeError
```

Reply:
The problem is in `np_scalar`. `Instruction does not dominate all uses!` means that not all controlflow paths have defined that the value before reaching the use-site of it. 
The easiest way to solve this is to use `alloca` (e.g. `cgutils.alloca_once`) to setup a stack slot where you can `store` the value inside the if branch; and later `load` the value outside the if-branch. You can see plenty of example in other boxing/unboxing code.